### PR TITLE
pixelnuke: init 2019-05-19

### PR DESCRIPTION
**Pull request by mrVanDalo**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/pixelnuke/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub, libevent, glew, glfw, ... }:
+
+stdenv.mkDerivation {
+
+  pname = "pixelnuke";
+
+  version = "2019-05-19";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/pixelnuke/default.nix:7]:
```suggestion
stdenv.mkDerivation {
  pname = "pixelnuke";
  version = "unstable-2019-05-19";
```

File: pkgs/applications/graphics/pixelnuke/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    repo = "pixelflut";
+    rev = "3458157a242ba1789de7ce308480f4e1cbacc916";
+    sha256 = "03dp0p00chy00njl4w02ahxqiwqpjsrvwg8j4yi4dgckkc3gbh40";
+  };
+
+  buildInputs = [ libevent glew glfw ];
+  buildPhase = ''
+    cd pixelnuke
+    make
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/pixelnuke/default.nix:20]:
```suggestion
  preBuild = ''
    cd pixelnuke
  '';
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Missing top level entry.

File: pkgs/applications/graphics/pixelnuke/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./pixelnuke $out/bin/
+  '';
+
+  meta = with lib; {
+    description = "Pixelflut: Multiplayer canvas";
```
Comment (OPNA2608) [inline pkgs/applications/graphics/pixelnuke/default.nix:28]: Doesn't fit [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/b5b66b1c80409e6e86dd9a3623f98edc4ece8020/.github/CONTRIBUTING.md):

> - `meta.description` should:
> …
>   - **Not start with the package name.**

`pixelnuke` is only one of three server implementations in the `pixelflut` repository, so I think I'd also prefer a mention of this in the description.

```suggestion
    description = "Multiplayer canvas (C implementation)";
```

File: pkgs/applications/graphics/pixelnuke/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub, libevent, glew, glfw, ... }:
```
Comment (OPNA2608) [inline pkgs/applications/graphics/pixelnuke/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, libevent, glew, glfw }:
```

File: pkgs/applications/graphics/pixelnuke/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  buildInputs = [ libevent glew glfw ];
+  buildPhase = ''
+    cd pixelnuke
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./pixelnuke $out/bin/
```
Comment (OPNA2608) [inline pkgs/applications/graphics/pixelnuke/default.nix:24]:
```suggestion
    install -Dm755 ./pixelnuke $out/bin/pixelnuke
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/pixelnuke/default.nix (hunk 2 above)
Comment (OPNA2608) [reply pkgs/applications/graphics/pixelnuke/default.nix:20]: Or

```suggestion
  sourceRoot = "source/pixelnuke";
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
> Missing top level entry.

This is still a problem. You need to make an entry for this derivation in `pkgs/top-level/all-packages.nix`, otherwise it essentially doesn't exist as a package in Nixpkgs. The file is organized into categories (search for `###`) and more-or-less alphabetically-ordered in those categories. Please pick a position suited for its type of application and ordered alphabetically-ish and add the following there:

```nix
  pixelnuke = callPackage ../applications/graphics/pixelnuke { };
```

Commit this change as well, then squash all of your commits into one `pixelnuke: init at unstable-2019-05-19` commit.

**Comment by mrVanDalo** [comment]
Thank you guys, I always forget to put the package into top-level. Also did not know we have categories now. Should be all good now.

File: pkgs/applications/graphics/pixelnuke/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    owner = "defnull";
+    repo = "pixelflut";
+    rev = "3458157a242ba1789de7ce308480f4e1cbacc916";
+    sha256 = "03dp0p00chy00njl4w02ahxqiwqpjsrvwg8j4yi4dgckkc3gbh40";
+  };
+
+  buildInputs = [ libevent glew glfw ];
+  sourceRoot = "source/pixelnuke";
```
Comment (OPNA2608) [inline pkgs/applications/graphics/pixelnuke/default.nix:15]: You could put another newline between these two.

```suggestion
  buildInputs = [ libevent glew glfw ];
  
  sourceRoot = "source/pixelnuke";
```

But it's not really important to me.

**Review by OPNA2608 — APPROVED** [review]
Result of `nixpkgs-review pr 119709` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pixelnuke</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 119709` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pixelnuke</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -23577,6 +23577,8 @@ in
       '' + (drv.postInstall or "");
     });
 
+  pixelnuke = callPackage ../applications/graphics/pixelnuke {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:23580]:
```suggestion
  pixelnuke = callPackage ../applications/graphics/pixelnuke { };
```

**Pull request merged**